dewy/migrations/0001_schema.sql

```diff
@@ -3,13 +3,11 @@
 CREATE TYPE distance_metric AS ENUM ('cosine', 'l2', 'ip');
 
 CREATE TABLE collection (
-    id SERIAL NOT NULL,
-    name VARCHAR NOT NULL,
+    id VARCHAR NOT NULL,
```

Comment:
I think we should stick with a numeric or UUID (essentially numeric) primary key for purposes of joining and other things. I'd be more in favor of changing the API to lookup the collection ID by name in all cases.

I worry about the unintended implications of this change:

1. In a multi-tenant world, using collection as ID means that you can't support different tenants using the same name. Not desirable.
2. In different databases, supporting named IDs may be more or less difficult.

I would instead leave the schema as is, maybe add an index on the `name` column, and update the SQL queries used to implement the routes by taking a name and selecting on the name rather than the `id` column.

Reply:
This was my first though too, and I started implementing along these lines. The issue is that `collection_id` shows up in several different places, for example foreign keys on chunks and documents. In order to retain the current identifier we'd either need to hide this as an implementation detail, accept a string in all those places, or require users to map from name to identifier.

I don't think this change prevents multiple tenants from using the same name; it just means we'd use a composite primary key (tenant + collection) and do joins conditional on both collection ID and tenant ID - neither of which seems like a problem. 

WRT different DB's, this isn't an issue for either of the two we're focusing on (Postgres and C*) - which DB's would cause problems for this schema?

My general feeling is that there's not much point in exposing both ID and Name to the user, and if we're not going to expose them I don't see any technical reason to keep them in the implementation - it just adds complexity.